modules/features2d/src/orb.cpp

```diff
@@ -995,15 +1008,22 @@ void ORB_Impl::detectAndCompute( InputArray _image, InputArray _mask,
         // In short, ultimately the descriptor should
         // ignore octave parameter and deal only with the keypoint size.
         nLevels = 0;
+        int prevLevel = firstLevel;
         for( i = 0; i < nkeypoints; i++ )
         {
-            level = keypoints[i].octave;
-            CV_Assert(level >= 0);
-            if( i > 0 && level < keypoints[i-1].octave )
+            level = getLevel(keypoints[i]);
+            if (level < 0) //issue #10555
```

Comment:
@vpisarev As you can see here, the negative level value is used to adjust the `firstLevel` offset.